firefox_ui_harness/runtests.py

```diff
@@ -29,9 +48,46 @@ def __init__(self, *args, **kwargs):
         self.test_handlers = [FirefoxTestCase]
 
 
-def run():
-    cli(runner_class=ReleaseTestRunner, parser_class=ReleaseTestParser)
+def startTestRunner(runner_class, options, tests):
+    try:
+        # If the specified binary is an installer it needs to be installed
+        installed = False
+        if mozinstall.is_installer(options.binary):
```

Comment:
It was working fine for us all the years long. But if you think that is better, tell me and I will do it. If we do it, then they are exclusive.